.github/workflows/base.yml

```diff
@@ -48,6 +48,14 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      # Install pyright
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - name: Install pyright
+        run: npm install -g pyright@1.1.223
+
```

Comment:
I would prefer that this is done in the `noxfile.py` so that it runs also locally. I never modify the github actions workflow, it just executes things that are declared in the nox sessions.

For this, you probably simply have to edit the "tests" nox session. Note: you may have to look at the value of the session parameter for python version in order to only install it on the appropriate python versions.

If you have trouble finding how let me know, I'll have a look